web/packages/teleterm/src/services/appUpdater/appUpdater.ts

```diff
@@ -284,6 +288,26 @@ export class AppUpdater {
     await this.checkForUpdates();
   }
 
+  /**
+   * Removes the managing cluster if it matches the given cluster URI.
+   * Cancels any in-progress update that was triggered by this cluster.
+   */
+  async maybeRemoveManagingCluster(clusterUri: RootClusterUri): Promise<void> {
+    const { managingClusterUri } = this.storage.get();
+    if (managingClusterUri === clusterUri) {
+      this.storage.put({ managingClusterUri: undefined });
+    }
+
+    // checkForUpdates will discard any update triggered by the removed managing
+    // cluster. If a different update is found, do not download it automatically.
+    // Currently, updates aren't checked in the background, and on Windows and Linux,
+    // users may be surprised by an admin prompt when there is an update to install
+    // after quitting the app.
+    // We may revisit this behavior in the future. For example, if we introduce
+    // a UI notification indicating that there's an update to be installed.
+    await this.checkForUpdates({ noAutoDownload: true });
```

Comment:
Is there any way we can create tests for this? It feels quite brittle, as in we don't directly discard any updates here but rather we depend on `this.checkForUpdates` doing this somewhere underneath when called with `noAutoDownload`. It'd be at least nice to cover that it in fact does this when `noAutoDownload` is used.

Is it possible to mock/fake `autoUpdater`? The logic in this file is already rather complex and we just don't have any tests for it, meaning that even this edge case here would need to be covered by the test plan.

Reply:
I thought it would a pain to mock `autoUpdater`, but it wasn't that bad! 
I added tests for updates auto download and discarding them when the latest check returns a new update or no update, as this part feels the most brittle.